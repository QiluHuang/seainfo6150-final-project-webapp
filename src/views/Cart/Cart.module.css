.cart_container {
    margin: 40px;
}

.title_row {
    display: flex;
    justify-content: space-between;
}

.container {
    width: auto;
    height: 100%;
    font-size: 1.2em;
  }
  

.list_row {
  list-style-type: decimal;
}  


.wrapper {
  height: 100%;
  display: flex;
  flex-direction: column;
  /* justify-content: space-between; */
  padding-left: 0px;
  text-align: center;
  font-family: Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;
  
}
 
  
.upper_image {
    text-align: center;
    padding: 10px;
    /* padding-top: 10px;
    padding-bottom: 10px; */
    width: 40%;
}

.index {
  width: 20%;
  font-size: 1.2em;
}

.dishName {
  width: 40%;
  font-size: 1.2em;
}

.quantity {
  width: 20%;
  font-size: 1.2em;
}

.price {
  width: 10%;
  font-size: 1.2em;
}


.bottom_row {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    margin-right: 100px;
}

.button {
  float: right;
  margin-right: 100px;
  margin-bottom: 100px;
}


