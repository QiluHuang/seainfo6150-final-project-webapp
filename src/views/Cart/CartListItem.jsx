import React, { useState } from "react";
import { Switch, Route, Link, useRouteMatch } from "react-router-dom";

import PropTypes from "prop-types";
import styles from "./CartListItem.module.css";
import DishImageMedium from "../../components/Images/DishImageMedium/DishImageMedium.jsx";

const CartListItem = (props) => {
    // console.log(props);
  return (
      <div className={styles.container}>
          <div className={styles.wrapper}>
              <div className={styles.upper_image}>
                  <DishImageMedium url={props.dish.image._url} title={props.dish.name}/>
              </div>
              <div className={styles.dishName}>{props.dish.name}</div>
              <div className={styles.quantity}>1</div>
              <div className={styles.price}>{props.dish.price}</div> 
          
                 
                                   
          </div>
        

      </div>
  );
};

CartListItem.propTypes = {
  dish: PropTypes.object.isRequired,
//   onClick: PropTypes.func.isRequired
};
export default CartListItem;