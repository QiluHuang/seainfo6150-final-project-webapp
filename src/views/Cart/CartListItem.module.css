.container {
    font-family: Arial, Helvetica, sans-serif;
    vertical-align: middle;
}

.wrapper {
    display: flex;
    flex-direction: row;
    vertical-align: middle;
    /* justify-content: space-between; */
}

.upper_image {
    margin-left: 100px;
    width: 25%;
}

.dishName {
    width: 38%;
    font-size: 1.2em;
}

.quantity {
    width: 22%;
    font-size: 1.2em;
}

.price {
    /* width: 10% */
    font-size: 1.2em;
}


.upper_image {
    /* width: 200px; */
    height: auto;
    text-align: center;
    padding: 10px;
}






@media (max-width: 768px) {
    .upper_image {
        display: none;
    }

    .dishName {
        width: 70%;
        font-size: 1em;
    }

    .quantity {
        width: 25%;
        font-size: 1em;

    }
  }
  
  @media (min-width: 768px) and (max-width: 1390px) {
    .dishName {
        width: 30%;
    }
  }