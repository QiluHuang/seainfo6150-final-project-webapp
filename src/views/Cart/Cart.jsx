import React, { useState, useEffect } from "react";
import { Switch, Route, Redirect, NavLink, Link, useRouteMatch } from "react-router-dom";
import styles from './Cart.module.css';
import PropTypes from "prop-types";


import Button from "../../components/Buttons/Button/Button";
import CartListItem from "./CartListItem.jsx";

const Cart = () => {
  // console.log(props);

    let [fetchedData, setFetchedData] = useState([]);

    useEffect(() => {
        let fetchData = async () => {
          // performs a GET request
          let response = await fetch("https://demo1273150.mockable.io/cover");
          // let response = await fetch(`${dataUrl}`);
          let responseJson = await response.json();
          setFetchedData(Object.values(responseJson));
        };
    
        if (!fetchedData.length) {
          fetchData();
        }
      }, [fetchedData]);
    
      let displayContent;
    
      // console.log(fetchedData);
  
  
      if (fetchedData.length) {
    
        displayContent = (
          <div className={styles.container}>
              <div>
                  <ul className={styles.castList_container}>
                      {fetchedData.map((dish, index) => (
                              <li className={styles.list_row} key={index}>
                                      <CartListItem dish={dish} key={dish.slug} />
                              </li>
                          ))}
                  </ul>
              </div>
          </div>
        );
      } else {
        displayContent = <div>You didn't put anything into the Cart!</div>;
      }

    return (
        <div>
                <div className={styles.cart_container}>
                    <div className={styles.title_row}>
                        <div className={styles.index}><h4>Index</h4></div>
                        <div className={styles.dishName}><h4>Name</h4></div>
                        <div className={styles.quantity}><h4>Quantity</h4></div>
                        <div className={styles.price}><h4>Price</h4></div>
                    </div>
                    {displayContent}
                    <div className={styles.bottom_row}>
                        <div><h4>Total: </h4></div>
                        <div><h4>$88.00</h4></div>
                    </div>
                    <div className={styles.button}>
                      <Link to="/checkout">
                        <Button buttonText="Checkout"/>
                      </Link>
                    </div>
                </div>
        </div>

  )
}

// Cart.propTypes = {
//   dishes: PropTypes.array.isRequired,
// };

export default Cart;
