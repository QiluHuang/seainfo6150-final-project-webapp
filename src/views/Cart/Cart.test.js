import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Cart from "./Cart.jsx";

describe("Cart tests", () => {
    it("renders the cart component correctly", () => {
    const { container } = render(<Router><Cart /></Router>);
    expect(container).toMatchSnapshot();
    });
});