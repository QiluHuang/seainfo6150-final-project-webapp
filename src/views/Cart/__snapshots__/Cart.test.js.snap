// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart tests renders the cart component correctly 1`] = `
<div>
  <div>
    <div
      class="cart_container"
    >
      <div
        class="title_row"
      >
        <div
          class="index"
        >
          <h4>
            Index
          </h4>
        </div>
        <div
          class="dishName"
        >
          <h4>
            Name
          </h4>
        </div>
        <div
          class="quantity"
        >
          <h4>
            Quantity
          </h4>
        </div>
        <div
          class="price"
        >
          <h4>
            Price
          </h4>
        </div>
      </div>
      <div>
        You didn't put anything into the Cart!
      </div>
      <div
        class="bottom_row"
      >
        <div>
          <h4>
            Total: 
          </h4>
        </div>
        <div>
          <h4>
            $88.00
          </h4>
        </div>
      </div>
      <div
        class="button"
      >
        <a
          href="/checkout"
        >
          <button
            class="button"
          >
            Checkout
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
